src/solver/optimisation/opt_appel_solveur_lineaire.cpp

```diff
@@ -226,21 +226,24 @@ static SimplexResult OPT_TryToCallSimplex(
     LinearProblemBuilder linearProblemBuilder(legacyLinearProblem);
     if (options.useOrtools)
     {
-        auto filler = std::make_shared<LegacyLinearProblemFillerImpl>(&Probleme); // TODO: merge this with LegacyLinearProblemImpl ?
+        auto filler = std::make_shared<LegacyLinearProblemFillerImpl>(
+          &Probleme); // TODO: merge this with LegacyLinearProblemImpl ?
         linearProblemBuilder.addFiller(filler);
         // TODO: we can add extra fillers here
         for (const auto& additionalFiller : gAdditionalFillers)
             linearProblemBuilder.addFiller(additionalFiller);
 
         // sinon renvoyer le builder ou le problem à une autre classe
         // Required for the balance constraint indices
-        gLinearProblemData.legacy.constraintMapping = &problemeHebdo->CorrespondanceCntNativesCntOptim;
+        gLinearProblemData.legacy.constraintMapping
+          = &problemeHebdo->CorrespondanceCntNativesCntOptim;
         gLinearProblemData.legacy.areaNames = &problemeHebdo->NomsDesPays;
         // TODO : add data here
-        linearProblemBuilder.build(gLinearProblemData);
+        linearProblemBuilder.build(gLinearProblemData[problemeHebdo->year]);
```

Comment:
Assuming `gLinearProblemData` contains the appropriate data, scenarization of new components should work natively for legacy problems.